starknet_py/net/models/transaction.py

```diff
@@ -1,28 +1,245 @@
+import marshmallow
+import marshmallow_dataclass
+from marshmallow import fields
+from starkware.starknet.services.api.contract_class import ContractClass
+from starkware.starknet.services.api.gateway.transaction_utils import (
+    compress_program_post_dump,
+    decompress_program,
+)
 
 from starknet_py.cairo.selector import get_selector_from_name
+from starknet_py.net.models import compute_address
 from starknet_py.net.models.chains import StarknetChainId
+from starknet_py.net.schemas.common import Felt, NoneFelt
 from starknet_py.utils.crypto.transaction_hash import (
     TransactionHashPrefix,
+    compute_declare_transaction_hash,
+    compute_deploy_account_transaction_hash,
     compute_transaction_hash,
 )
-from starknet_py.utils.docs import as_our_module
 
-Invoke = InvokeFunction = as_our_module(IF)
-Transaction = as_our_module(T)
-TransactionType = as_our_module(TT)
-Declare = as_our_module(DCL)
-DeployAccount = as_our_module(DAC)
+
+class TransactionType(Enum):
+    """
+    Enum representing transaction types.
+    """
+
+    DECLARE = 0
+    DEPLOY = auto()
+    DEPLOY_ACCOUNT = auto()
+    INITIALIZE_BLOCK_INFO = auto()
+    INVOKE_FUNCTION = auto()
+    L1_HANDLER = auto()
+
+
+# pylint: disable=no-self-use, unused-argument
+@dataclass(frozen=True)
+class Transaction:
+    """
+    StarkNet transaction base class.
+    """
+
+    version: int = field(metadata={"marshmallow_field": Felt()})
+
+    @classmethod
+    @abstractmethod
+    def tx_type(cls) -> TransactionType:
+        """
+        Returns the corresponding TransactionType enum.
+        """
+
+    @abstractmethod
+    def calculate_hash(self, chain_id: StarknetChainId) -> int:
+        """
+        Calculates the transaction hash in the StarkNet network - a unique identifier of the
+        transaction. See compute_transaction_hash() docstring for more details.
+        """
+
+
+@dataclass(frozen=True)
+class AccountTransaction(Transaction, ABC):
+    """
+    Represents a transaction in the StarkNet network that is originated from an action of an
+    account.
+    """
+
+    max_fee: int = field(metadata={"marshmallow_field": Felt()})
+    signature: List[int] = field(
+        metadata={"marshmallow_field": fields.List(fields.String())}
+    )
+    nonce: int = field(metadata={"marshmallow_field": Felt()})
+
+
+@dataclass(frozen=True)
+class Declare(AccountTransaction):
+    """
+    Represents a transaction in the StarkNet network that is a declaration of a StarkNet contract
+    class.
+    """
+
+    contract_class: ContractClass
+    # The address of the account contract sending the declaration transaction.
+    sender_address: int = field(metadata={"marshmallow_field": Felt()})
+
+    @classmethod
+    def tx_type(cls) -> TransactionType:
+        return TransactionType.DECLARE
+
+    @marshmallow.post_dump
+    def compress_program_post_dump(
+        self, data: Dict[str, Any], many: bool, **kwargs
```

Comment:
Are `kwargs` needed here?

Reply:
Unfortunately yes :/ `@marshmallow.post_dump` requires that